File: Examples/test-suite/java/template_template_parameters_runme.java


import template_template_parameters.*;

public class template_template_parameters_runme {

  static {
    try {
	System.loadLibrary("template_template_parameters");
    } catch (UnsatisfiedLinkError e) {
      System.err.println("Native code library failed to load. See the chapter on Dynamic Linking Problems in the SWIG Java documentation for help.\n" + e);
      System.exit(1);
    }
  }

  public static void main(String argv[]) {
    // Test first part
    ListFastBool listBool = new ListFastBool();
    listBool.setItem(true);
    boolean x_boolean = listBool.getAllotype();
    if (listBool.getItem() != true)
      throw new RuntimeException("Failed");

    ListDefaultDouble listDouble = new ListDefaultDouble();
    listDouble.setItem(10.2);
    double x_double = listDouble.getAllotype();
    if (listDouble.getItem() != 10.2)
      throw new RuntimeException("Failed");

    // Test second part
    FloatTestStruct floatTestStruct = new FloatTestStruct();
    FloatContainer2 floatContainer2 = floatTestStruct.getX();
    floatContainer2.setX(8.1f);
    IntTestStruct intTestStruct = new IntTestStruct();
    IntContainer1 intContainer1 = intTestStruct.getX();
    intContainer1.setX(91);
    if (intContainer1.getX() != 91)
      throw new RuntimeException("Failed");
    if (intTestStruct.getX().getX() != 91)
      throw new RuntimeException("Failed");
    IntTestStruct intTestStructReturned = template_template_parameters.TestStructContainer1Method(intTestStruct);
    if (intTestStructReturned.getX().getX() != 101)
      throw new RuntimeException("Failed");
  }
}

File: Examples/test-suite/kotlin/template_template_parameters_runme.kt
@file:JvmName("template_template_parameters_runme")

import template_template_parameters.*

fun main() {
    try {
        System.loadLibrary("template_template_parameters")
    } catch (e: UnsatisfiedLinkError) {
        System.err.println("Native code library failed to load. See the chapter on Dynamic Linking Problems in the SWIG Java documentation for help.\n" + e)
        kotlin.system.exitProcess(1)
    }

    // Test part 1
    val listBool = ListFastBool()
    listBool.item = true
    val x_boolean = listBool.allotype
    if (listBool.item != true)
        throw RuntimeException("Failed")

    val listDouble = ListDefaultDouble()
    listDouble.item = 10.2
    val x_double = listDouble.allotype
    if (listDouble.item != 10.2)
        throw RuntimeException("Failed")

    // Test part 2
    val floatTestStruct = FloatTestStruct()
    val floatContainer2 = floatTestStruct.x!!
    floatContainer2.x = 8.1f
    val intTestStruct = IntTestStruct()
    val intContainer1 = intTestStruct.x!!
    intContainer1.x = 91
    if (intContainer1.x != 91)
        throw RuntimeException("Failed")
    if (intTestStruct.x!!.x != 91)
        throw RuntimeException("Failed")
    val intTestStructReturned = template_template_parameters.TestStructContainer1Method(intTestStruct)
    if (intTestStructReturned.x!!.x != 101)
        throw RuntimeException("Failed")

    // Test part 3
    val mfi99 = MyFootInt99()
    mfi99.OperatorPlusEquals(mfi99)
}
